app.rb

```diff
@@ -12,26 +12,96 @@ def initialize
     @rentals = []
   end
 
-  def user_input(text)
-    print text
-    gets.chomp
+  class CreateList
+    def user_input(text)
+      print text
+      gets.chomp
+    end
   end
 
   def list_all_books
-    if @books.empty?
+    ListBook.new.list_all_books(@books)
+  end
+
+  def list_all_people
+    ListPeople.new.list_all_people(@people)
+  end
+
+  def create_a_person
+    CreateList.new.create_a_person(@people)
+  end
+
+  def create_a_book
+    CreateList.new.create_a_book(@books)
+  end
+
+  def create_a_rental
+    CreateList.new.create_a_rental(@books, @people, @rentals)
+  end
+
+  def list_all_rentals
+    ListRental.new.list_all_rentals(@rentals, @people)
+  end
+
+  def operation(input)
+    case input
+    when 1
+      list_all_books
+    when 2
+      list_all_people
+    when 3
+      create_a_person
+    when 4
+      create_a_book
+    when 5
+      create_a_rental
+    when 6
+      list_all_rentals
+    else
+      puts 'Please choose a valid option number'
+    end
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    loop do
+      puts '*' * 60
+      puts '
+          1- List all books
+          2- List all people
+          3- Create a new person
+          4- Create a new book entry
+          5- Create a new rental entry
+          6- List all rentals for a given person id
+          7- Quit'
+
+      input = user_input('Choose an option: ').to_i
+
+      break if input == 7
+
+      operation(input)
+    end
+  end
+end
+
+class ListBook
+  def list_all_books(books)
+    if books.empty?
       puts 'There is no book added'
     else
-      @books.each_with_index do |book, index|
+      books.each_with_index do |book, index|
         puts "(#{index + 1})  Title: #{book.title}, Title: #{book.author}"
       end
     end
   end
+end
```

Comment:
- [ ] Great job so far but I think it is not a good practice to put each single method in seprate class since you can put them in one class I see in the requirements to `Extract some pieces of logic into separate class(es).` but I think not like this to put each method in a seprate class may be that requirements not clear 100% but this to make main in one file and app in another file contain class with all the methods but you still should refactor the methods and its validation 🙏🏻✅🔥💯🚀